app/src/main/java/org/oppia/app/customview/interaction/NumericInputInteractionView.kt

```diff
@@ -0,0 +1,20 @@
+package org.oppia.app.customview.interaction
+
+import android.content.Context
+import android.util.AttributeSet
+import android.widget.EditText
+import org.oppia.app.model.InteractionObject
+
+/** The custom EditText class for numeric input interaction view. */
+class NumericInputInteractionView @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyle: Int = android.R.attr.editTextStyle
+) : EditText(context, attrs, defStyle), InteractionAnswerRetriever {
+
+  override fun getPendingAnswer(): InteractionObject {
+    return if (text.isNullOrEmpty()) (InteractionObject.newBuilder().build()) else (InteractionObject.newBuilder().setReal(
```

Comment:
I think this could be cleaned up a bit to be slightly clearer. Consider two possibilities:

```kotlin
return if (!text.isNullOrEmpty()) {
  InteractionObject.newBuilder.setReal(text.toString().toDouble()).build()
} else {
  InteractionObject.getDefaultInstance()
}
```

Or:

```kotlin
val interactionObjectBuilder = InteractionObject.newBuilder()
if (!text.isNullOrEmpty()) {
  interactionObjectBuilder.setReal(text.toString().toDouble())
}
return interactionObjectBuilder.build()
```

Reply:
resolved